File: examples/api/python/create_visibility_run.py
import json
import os
import urllib.request


api_key = os.environ.get("SLEEPWALKER_API_KEY")
if not api_key:
    raise SystemExit("Set SLEEPWALKER_API_KEY first.")

payload = json.dumps({
    "url": "https://www.sleepwalker.ai",
    "target_entity": "Sleepwalker",
    "prompts": ["What are the best AI visibility tools?"],
    "platforms": ["perplexity"],
    "idempotency_key": "visibility-sleepwalker-example-001",
}).encode("utf-8")

request = urllib.request.Request(
    "https://api.sleepwalker.ai/v1/visibility/runs",
    data=payload,
    headers={
        "Authorization": f"Bearer {api_key}",
        "Content-Type": "application/json",
    },
    method="POST",
)

with urllib.request.urlopen(request, timeout=60) as response:
    print(json.dumps(json.load(response), indent=2))

File: examples/api/javascript/create-visibility-run.mjs
const apiKey = process.env.SLEEPWALKER_API_KEY;

if (!apiKey) {
  throw new Error("Set SLEEPWALKER_API_KEY first.");
}

const response = await fetch("https://api.sleepwalker.ai/v1/visibility/runs", {
  method: "POST",
  headers: {
    Authorization: `Bearer ${apiKey}`,
    "Content-Type": "application/json",
  },
  body: JSON.stringify({
    url: "https://www.sleepwalker.ai",
    target_entity: "Sleepwalker",
    prompts: ["What are the best AI visibility tools?"],
    platforms: ["perplexity"],
    idempotency_key: "visibility-sleepwalker-example-001",
  }),
});

if (!response.ok) {
  throw new Error(`Sleepwalker API error: ${response.status} ${await response.text()}`);
}

console.log(JSON.stringify(await response.json(), null, 2));
